server/app.js

```diff
@@ -0,0 +1,250 @@
+  'options', 'head', 'trace',
+  'connect', 'all', 'copy', 'link',
+  'unlink', 'purge', 'lock', 'unlock',
+  'propfind', 'view',
+];
+
+/**
+ * Handles the addition of the given path to the app's swagger schema
+ * @param {string} path The path to add
+ * @param {string} method The HTTP method,
+ * could also be 'use' or 'all' in which case we default to 'get'
+ * @param {Object} [options] Optional swagger definition that will descriptions of the given path.
+ * @returns {void}
+ */
+const addToAppSchema = (path, method, options) => {
+  const otherOptions = options || {};
+  app.schema.paths[path] = {
+    [httpMethods.includes(method) ? method : 'get']: {
+      responses: {},
+      ...otherOptions,
+    },
+    ...app.schema.paths[path],
+  };
+};
+
+/**
+ * Checks whether the given instance is an instance of express.Router.
+ * @param {*} instance The instance to perform the check against.
+ * @return {boolean} true if instance is an instance of express.Router, otherwise false.
+ */
+const isRouter = instance => Object.getPrototypeOf(instance) === express.Router;
+
+/**
+ * Handles the case when an instance of express.Router is given as the last argument to an app's
+ * method
+ * @param {string} definedPath The path for which this router should handle
+ * @param {express.Router} router The router which was given
+ * @param {Object} [options] Optional swagger definition that will descriptions of the given path.
+ * @return {Promise<*>} A Promise that resolves when the express Router has been handled
+ */
+const handleExpressRouter = (definedPath, router, options) => loopSequentially(router.stack,
+  (layer) => {
+    // If the router was given a router,
+    // get the path it handles from the regex expression as a string,
+    // append it to the definedPath,
+    // then call this function recursively with it as the first argument,
+    // it's handle as the second argument, and the given options as the third
+    if (layer.name === 'router') {
+      const regexPath = layer.regexp.toString().split('\\')[1];
+      return handleExpressRouter(`${definedPath}${regexPath}`, layer.handle, options);
+    }
+
+    if (!layer.route) {
+    // This means that .use() was used instead of a HTTP method and the last argument is
+    // a request handler of signature (req, res, next) => {}
+    // For documentation purposes, we will use the get method to add it to swagger
+    // because we will not have access to any HTTP method
+      const regexPath = layer.regexp.toString().split('\\')[1];
+      const path = `${definedPath}${regexPath}`;
+      return addToAppSchema(path, 'get', options);
+    }
+
+    // This means HTTP methods were used on the router and therefore we have access
+    // to them and the route that they handle, so we add this definition to the swagger
+    // schema for each method
+    return loopSequentially(Object.keys(layer.route.methods), (routerMethod) => {
+      const path = `${definedPath}${layer.route.path}`;
```

Comment:
@davenjeru using the name `path` here may conflict with the imported path module

Reply:
@davenjeru I think you would rather rename the import since you have used the variable `path` in a lot of places.

Reply:
Okay...